db/db_impl.cc

```diff
@@ -1177,10 +1177,11 @@ Status DBImpl::GetImpl(const ReadOptions& read_options,
   return s;
 }
 
-std::vector<Status> DBImpl::MultiGet(
-    const ReadOptions& read_options,
-    const std::vector<ColumnFamilyHandle*>& column_family,
-    const std::vector<Slice>& keys, std::vector<std::string>* values) {
+void DBImpl::MultiGet(const ReadOptions& read_options,
+                      const std::vector<ColumnFamilyHandle*>& column_family,
+                      const std::vector<Slice>& keys,
+                      std::vector<PinnableSlice>* values,
+                      std::vector<Status>& statuses) {
```

Comment:
I was wrong about an allocation for `values`. I saw your comment somewhere else that that users are responsible to preallocate an array with the same length. This clear contract needs to be clearly documented before this function.